platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java

```diff
@@ -108,60 +183,66 @@ public void addSession(HttpSession session) {
         sessions = new HashSet<>();
-        this.sessions.put(id, sessions);
+        sessionMap.put(id, sessions);
       } else {
-        // Check for session already in cluster, copy over session information to new session
-        Iterator<WeakReference<HttpSession>> iterator = sessions.iterator();
-        if (iterator.hasNext()) {
-          WeakReference<HttpSession> weakReference = iterator.next();
-          if (weakReference != null) {
-            HttpSession httpSession = weakReference.get();
-            if (httpSession != null) {
-              Enumeration enumeration = httpSession.getAttributeNames();
-              while (enumeration.hasMoreElements()) {
-                Object obj = enumeration.nextElement();
-                if (obj instanceof String) {
-                  Object value = httpSession.getAttribute((String) obj);
-                  if (value != null) {
-                    session.setAttribute((String) obj, value);
-                  }
-                }
-              }
-              session.setMaxInactiveInterval(httpSession.getMaxInactiveInterval());
-            }
-          }
-        }
+        copySessionInfo(session, sessions);
       }
       sessions.add(ref);
     }
   }
 
+  @SuppressWarnings("squid:S2441" /* Value stored in session will not be serialized */)
+  private void copySessionInfo(HttpSession session, Set<WeakReference<HttpSession>> sessions) {
+    // Check for session already in cluster, copy over session information to new session
+    final HttpSession httpSession =
+        Optional.of(sessions.iterator())
+            .filter(Iterator::hasNext)
+            .map(Iterator::next)
+            .map(Reference::get)
+            .orElse(null);
+
+    if (httpSession == null) {
+      return;
+    }
+
+    Enumeration enumeration = httpSession.getAttributeNames();
+    while (enumeration.hasMoreElements()) {
+      Object obj = enumeration.nextElement();
+      if (obj instanceof String) {
+        Object value = httpSession.getAttribute((String) obj);
+        if (value != null) {
+          session.setAttribute((String) obj, value);
+        }
+      }
+    }
+    session.setMaxInactiveInterval(httpSession.getMaxInactiveInterval());
+  }
+
   /** @see org.eclipse.jetty.server.SessionIdManager#removeSession(HttpSession) */
   @Override
   public void removeSession(HttpSession session) {
     String id = getClusterId(session.getId());
 
     synchronized (this) {
-      Collection<WeakReference<HttpSession>> sessions = this.sessions.get(id);
+      Collection<WeakReference<HttpSession>> sessions = sessionMap.get(id);
       if (sessions != null) {
         Iterator<WeakReference<HttpSession>> iter = sessions.iterator();
         while (iter.hasNext()) {
           WeakReference<HttpSession> ref = iter.next();
           HttpSession s = ref.get();
           if (s == null) {
             iter.remove();
-            continue;
           }
           if (s == session) {
```

Comment:
❓ Do we want to check equality here with .equals or is this really meant to compare references ?

Reply:
Instance equality is correct here.